### PR TITLE
Add simple retarget example

### PR DESCRIPTION
**Pull request by JonatanAntoni**

### CODE REVIEW

File: ARM.CMSIS-Compiler.pdsc [hunk 1]
```diff
@@ -482,4 +482,21 @@
...
   </components>
+
+  <examples>
+    <example name="retarget" folder="example" doc="README.md">
+      <description>Simple example retargeting stdout to USART.</description>
+      <board name="V2M-MPS2" vendor="ARM"/>
+      <project>
+        <environment name="cmsis" load="retarget.csolution.yml"/>
+      </project>
+      <attributes>
```
Comment (ReinhardKeil) [inline ARM.CMSIS-Compiler.pdsc:493]: I suggest to remove <attributes> from the examples.

**Review by ReinhardKeil — APPROVED** [review]

File: example/retarget.csolution.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/Open-CMSIS-Pack/devtools/schemas/projmgr/1.5.0/tools/projmgr/schemas/csolution.schema.json
```
Comment (jkrech) [inline example/retarget.csolution.yml:1]: It was suggested that we would no longer maintain the `# yaml-language-server: ...` line in our project files.

File: example/retarget.cproject.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/Open-CMSIS-Pack/devtools/schemas/projmgr/1.5.0/tools/projmgr/schemas/cproject.schema.json
```
Comment (jkrech) [inline example/retarget.cproject.yml:1]: It was suggested that we would no longer maintain the `# yaml-language-server: ...` line in our project files.

File: ARM.CMSIS-Compiler.pdsc (hunk 1 above)
Comment (JonatanAntoni) [reply ARM.CMSIS-Compiler.pdsc:493]: You are talking about removing the `<attributes>`?

File: ARM.CMSIS-Compiler.pdsc [hunk 4]
```diff
@@ -482,4 +482,21 @@
       </files>
     </component>
   </components>
+
+  <examples>
+    <example name="retarget" folder="example" doc="README.md">
+      <description>Simple example retargeting stdout to USART.</description>
+      <board name="V2M-MPS2" vendor="ARM"/>
+      <project>
+        <environment name="cmsis" load="retarget.csolution.yml"/>
```
Comment (JonatanAntoni) [inline ARM.CMSIS-Compiler.pdsc:491]: We don't have this in any IDE, yet.
@jkrech, shall we go with `name="cmsis"`?

**Review by KeilChris — APPROVED** [review]
Meaningful example running purely in simulation.

**Pull request merged**